pkg/services/authapi/authapi.go

```diff
@@ -0,0 +1,398 @@
+	LabelPolicies []LabelPolicy `json:"labelPolicies"`
+	Type          string        `json:"type"`
+}
+
+type LabelPolicy struct {
+	Selector string `json:"selector"`
+}
+
+type createAccessPolicyResponse struct {
+	Data AccessPolicy `json:"data"`
+}
+
+type AccessPolicy struct {
+	ID   string `json:"id"`
+	Name string `json:"name"`
+}
+
+type ListAccessPoliciesParams struct {
+	RequestID string
+	Name      string
+	// this is needed until we fully migrate from gcom to authapi
+	Region string
+}
+
+type listAccessPoliciesResponse struct {
+	Data []AccessPolicy `json:"data"`
+}
+
+type DeleteAccessPolicyParams struct {
+	RequestID      string
+	AccessPolicyID string
+	// this is needed until we fully migrate from gcom to authapi
+	Region string
+}
+
+type ListTokenParams struct {
+	RequestID        string
+	AccessPolicyName string
+	TokenName        string
+	// this is needed until we fully migrate from gcom to authapi
+	Region string
+}
+
+type CreateTokenParams struct {
+	RequestID string
+	// this is needed until we fully migrate from gcom to authapi
+	Region string
+}
+
+type CreateTokenPayload struct {
+	AccessPolicyID string    `json:"accessPolicyId"`
+	DisplayName    string    `json:"displayName"`
+	Name           string    `json:"name"`
+	ExpiresAt      time.Time `json:"expiresAt"`
+}
+
+type createTokenResponse struct {
+	Data Token `json:"data"`
+}
+
+// Token returned by gcom api when a token gets created.
+type Token struct {
+	ID             string `json:"id"`
+	AccessPolicyID string `json:"accessPolicyId"`
+	Name           string `json:"name"`
+	Token          string `json:"token"`
+}
+
+type DeleteTokenParams struct {
+	RequestID string
+	TokenID   string
+	// this is needed until we fully migrate from gcom to authapi
+	Region string
+}
+
+// TokenView returned by gcom api for a GET token request.
+type TokenView struct {
+	ID             string `json:"id"`
+	AccessPolicyID string `json:"accessPolicyId"`
+	Name           string `json:"name"`
+	DisplayName    string `json:"displayName"`
+	ExpiresAt      string `json:"expiresAt"`
+	FirstUsedAt    string `json:"firstUsedAt"`
+	LastUsedAt     string `json:"lastUsedAt"`
+	CreatedAt      string `json:"createdAt"`
+}
+
+type listTokensResponse struct {
+	Data []TokenView `json:"data"`
+}
+
+var _ Service = (*AuthApiClient)(nil)
+
+type AuthApiClient struct {
+	log        log.Logger
+	cfg        Config
+	httpClient *http.Client
+}
+
+type Config struct {
+	ApiURL string
+	Token  string
+}
+
+func New(cfg Config, httpClient *http.Client) Service {
+	return &AuthApiClient{
+		log:        log.New(LogPrefix),
+		cfg:        cfg,
+		httpClient: httpClient,
+	}
+}
+
+func (client *AuthApiClient) CreateAccessPolicy(ctx context.Context, params CreateAccessPolicyParams, payload CreateAccessPolicyPayload) (AccessPolicy, error) {
+	endpoint, err := url.JoinPath(client.cfg.ApiURL, "/v1/accesspolicies")
+	if err != nil {
+		return AccessPolicy{}, fmt.Errorf("building gcom access policy url: %w", err)
```

Comment:
Nit:
```suggestion
		return AccessPolicy{}, fmt.Errorf("building authapi access policy url: %w", err)
```
In the other methods that is the same as well.

Reply:
ouch! thanks! fixed in https://github.com/grafana/grafana/pull/96581/commits/72d342aa8fa2ab46cc9ad1abd65377159799b4bd